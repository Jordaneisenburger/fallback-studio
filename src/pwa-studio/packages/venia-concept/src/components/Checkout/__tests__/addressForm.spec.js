import React from 'react';
import testRenderer from 'react-test-renderer';
import AddressForm from '../addressForm';

jest.mock('../../../classify');

const mockCancel = jest.fn();
const mockSubmit = jest.fn();
const defaultProps = {
    cancel: mockCancel,
    submit: mockSubmit
};

beforeEach(() => {
    mockCancel.mockReset();
    mockSubmit.mockReset();
});

test('renders an AddressForm component', () => {
    const component = testRenderer.create(<AddressForm {...defaultProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders validation block with message if address is incorrect', () => {
    const props = {
        ...defaultProps,
        isAddressInvalid: true,
        invalidAddressMessage: 'whoops'
    };
    const component = testRenderer.create(<AddressForm {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('does not renders validation block with message if address is incorrect but no incorrect address message', () => {
    const props = {
        ...defaultProps,
        isAddressInvalid: true
    };
    const component = testRenderer.create(<AddressForm {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('cancel instance function calls props cancel function', () => {
    const { root } = testRenderer.create(<AddressForm {...defaultProps} />);

    const button = root.findAllByProps({ className: 'button' })[0];
    button.props.onClick();
    expect(mockCancel).toHaveBeenCalled();
});
