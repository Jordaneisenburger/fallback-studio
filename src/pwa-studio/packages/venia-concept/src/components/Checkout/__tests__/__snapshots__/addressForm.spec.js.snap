// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not renders validation block with message if address is incorrect but no incorrect address message 1`] = `
<form
  className="root"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="body"
  >
    <h2
      className="heading"
    >
      Shipping Address
    </h2>
    <div
      className="firstname"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          First Name
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="firstname"
              name="firstname"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="lastname"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Last Name
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="lastname"
              name="lastname"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="email"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Email
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="street0"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Street
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="street0"
              name="street[0]"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="city"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          City
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="city"
              name="city"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="region_code"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          State
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="region_code"
              name="region_code"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="postcode"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          ZIP
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="postcode"
              name="postcode"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="telephone"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Phone
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="telephone"
              name="telephone"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="validation"
    />
  </div>
  <div
    className="footer"
  >
    <button
      className="button"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="content"
      >
        Cancel
      </span>
    </button>
    <button
      className="button"
      type="submit"
    >
      <span
        className="content"
      >
        Use Address
      </span>
    </button>
  </div>
</form>
`;

exports[`renders an AddressForm component 1`] = `
<form
  className="root"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="body"
  >
    <h2
      className="heading"
    >
      Shipping Address
    </h2>
    <div
      className="firstname"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          First Name
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="firstname"
              name="firstname"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="lastname"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Last Name
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="lastname"
              name="lastname"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="email"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Email
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="street0"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Street
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="street0"
              name="street[0]"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="city"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          City
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="city"
              name="city"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="region_code"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          State
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="region_code"
              name="region_code"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="postcode"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          ZIP
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="postcode"
              name="postcode"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="telephone"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Phone
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="telephone"
              name="telephone"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="validation"
    />
  </div>
  <div
    className="footer"
  >
    <button
      className="button"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="content"
      >
        Cancel
      </span>
    </button>
    <button
      className="button"
      type="submit"
    >
      <span
        className="content"
      >
        Use Address
      </span>
    </button>
  </div>
</form>
`;

exports[`renders validation block with message if address is incorrect 1`] = `
<form
  className="root"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="body"
  >
    <h2
      className="heading"
    >
      Shipping Address
    </h2>
    <div
      className="firstname"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          First Name
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="firstname"
              name="firstname"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="lastname"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Last Name
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="lastname"
              name="lastname"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="email"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Email
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="street0"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Street
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="street0"
              name="street[0]"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="city"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          City
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="city"
              name="city"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="region_code"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          State
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="region_code"
              name="region_code"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="postcode"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          ZIP
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="postcode"
              name="postcode"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="telephone"
    >
      <div
        className="root"
      >
        <span
          className="label"
        >
          Phone
        </span>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              id="telephone"
              name="telephone"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="validation"
    >
      whoops
    </div>
  </div>
  <div
    className="footer"
  >
    <button
      className="button"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="content"
      >
        Cancel
      </span>
    </button>
    <button
      className="button"
      type="submit"
    >
      <span
        className="content"
      >
        Use Address
      </span>
    </button>
  </div>
</form>
`;
